core/gather/driver/execution-context.js

```diff
@@ -215,7 +223,13 @@ class ExecutionContext {
    */
   async evaluateOnNewDocument(mainFn, options) {
     const argsSerialized = ExecutionContext.serializeArguments(options.args);
-    const depsSerialized = options.deps ? options.deps.join('\n') : '';
+    const depsSerialized = (options.deps || []).map(dep => {
+      if (typeof dep === 'function') {
+        return `const ${dep.name} = ${dep}`;
+      } else {
+        return dep;
+      }
+    }).join('\n');
 
     const expression = `(() => {
```

Comment:
Does this need `esbuildFunctionNameStubString`? Could be related to the `lantern-idle-callback-long` failures since that shim is added on new document?

Reply:
It sounds reasonable, but we have multiple invocations of `evaluateOnNewDocument` so it might be tricky to verify that they *all* run on a new document. Maybe instead of checking on gotoUrl we check every time we evaluate code in the execution context?

The `lantern-idle-callback-long` smoke test might be our best way to verify this still.